### PR TITLE
Feature/scheduler profiles

### PR DESCRIPTION
**Pull request by nickve28**
Allows creation of `schedule_profiles` per account, which can be used to select specific images. 
Sub configs allow settings rules for matches, schedule_profiles allows for narrowing down results (compared to just the root dir path)
Use case: allow more control of scheduling specific groups

### CODE REVIEW

File: src/convert_pngs.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import os
```
Comment (nickve28) [inline src/convert_pngs.py]: move to bin folder

File: src/deviant_folders.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import sys
```
Comment (nickve28) [inline src/deviant_folders.py]: move to bin folder

File: src/utils/file_utils.py [hunk 3]
```diff
@@ -59,26 +65,51 @@ def rename_json_if_exists(filepath: str, new_filepath: str):
...
-        excluded_tags = [POSTED_TAG_MAPPING[platform] for platform in platforms]
+        excluded_tags.extend([POSTED_TAG_MAPPING[platform] for platform in account.platforms])
     if skip_queued:
-        excluded_tags.extend([QUEUE_TAG_MAPPING[platform] for platform in platforms])
-    return [f for f in files if not any(tag in f for tag in excluded_tags)]
+        excluded_tags.extend([QUEUE_TAG_MAPPING[platform] for platform in account.platforms])
+    return excluded_tags
+
+
+def exclude_files(files: List[str], account: Account, skip_queued: bool, skip_posted=True):
```
Comment (nickve28) [inline src/utils/file_utils.py]: needs tests and a possible rewrite

Comment (nickve28) [reply src/utils/file_utils.py]: note to self: 'flags' (INCLUDE_UNQUEUED | EXCLUDE_POSTED | ...)

**Review by alexopus — APPROVED** [review]

File: src/utils/file_utils.py [hunk 4]
```diff
@@ -59,26 +65,62 @@ def rename_json_if_exists(filepath: str, new_filepath: str):
...
+    if len(account.scheduler_profiles) == 0:
+        return False
+
+    for scheduler_profile in account.scheduler_profiles:
+        if any(matches_path(file, exclude_path) for exclude_path in scheduler_profile.exclude_paths):
+            return True
+    return False
+
+
+def is_excluded_file(account: Account, file: str, excluded_tags: List[str]):
```
Comment (nickve28) [inline src/utils/file_utils.py:100]: idea: turn each dir_path and excluded_dir path into 2 arrays, match on those
eg
pattern with excluded_tag # 1 
pattern with excluded tag # 2
excluded schedule profile # 1
 etc

**Pull request merged**